### DISCUSSION TITLE
When will the need for debug = True be fixed?

### DISCUSSION THREAD

**Discussion by rayw45** [kubilus1/autoortho#403] [General]
As it is the only way I, and many others I understand, can get AO 6.1 working in Windows 10 is to make the configuration file line debug = True rather than the default False.  This enables 6.1 to work without crashing XP12 with an unable to run error, but at a terribly high price of considerable delays and slow downs of course.  

So my question is, as this should be fairly clear to the developers what is causing this requirement for many of us,  when is this likely to be fixed ??  I love AO but really hate this terrible slow down caused by the necessary enabling of debugging.

To try to help isolate this issue, then perhaps others here, who also need this debug = True option, could share their set ups to see if some common cause can be identified.  So here are my system specs.

Windows 10 (with latest updates) 
Kaspersky AV security 
CPU - i5 10400F 
GFX card Aurus GTX 1080Ti 11 GB (with very latest regularly updated Nvidea drivers), moderately but safely overclocked
32Mb DDR Memory at 2666Mhz
Motherboard - Gigabyte H410M H V2 socket L1200 for 10th generation Intel cpus
Power supply 650W
X Plane latest version of 12.07 running at 1080 resolution
Using Dokan, but problem remains if changing to WinFSP
Various XP12 plugins including Lua's 3jFPS12, SAM 3 needed for some XP12 airport addons, AviTab, Airport Navigator, XChecklist.
A few XP12 third party aircraft mainly from Airfoilabs but problem remains using any aircraft including Laminar default.

**Comment by zippy77777** [comment]
I have managed to get it working after 2 days of trying. Dokan no longer works and winFSP does but they don't work together. Delete Dokan. If you use SAM, move the libraries into custom scenery folder manually (settings in xplane) and delete the rest of SAM from the custom scenery folder. Create symbolic links to the remaining library files. This will stop the AO files keep moving to the top of the scenery_packs.ini folder. It took a while experimenting but this eventually worked for me.

**Reply by Jeff2122** [reply]
Still wondering why some folks can't run without the debug=true flag. There has to be something in their Windows environment that only allows data traffic to flow if interrupted by writing a line of text. Makes no sense yet, but it will.

**Reply by malec** [reply]
I feel like it has to download everything in debug mode or doesn't use the cache. It takes a long time to render / load the scenery.

**Reply by Jeff2122** [reply]
I have Python installed on my system and I can run either the source code or the executable. I wonder what would happen if someone who has this debug problem could install python and try the source code version.
Does it matter if they run as administrator? That's been tried of course. 
Do these folks have RAID or any other kind of strange disk setup?
Intel vs AMD issue? i5 vs i7 vs i9? Motherboard brand? Time to start documenting/watching for trends?

**Reply by malec** [reply]
Just got it working by disabling sam, uninstalling dokan, using winfsp-2.0.23075, xplane 12, auto ortho 0.5.1-3. On intel i9, win 10, using win storage spaces for cache dir, added win defender exclusion on all auto ortho related dirs, xplane plane scenery dir, and xp process. Not sure if I have the patience to run from source 😅.

**Reply by zippy77777** [reply]
I can run with both True and False now. I also have the latest python script.

**Reply by Jeff2122** [reply]
There it is - a complete overhaul of the AO system is what @malec tried and succeeded. Can somebody else with the debug-true problem please follow his footsteps and see if this clears up the issue?

**Reply by rayw45** [reply]
Most of us now have AO 6.1 but I understand the same problem is with 5 too. Can malec try to upgrade to version 6.1 as it is supposed to be much better and faster than 5 anyway, that is as long as it works without debug switched on of course.  

I need SAM 3 for running a couple of quite expensive and much used beautifully created airports in Bangkok and Hong Kong, but use SAM for nothing else and am pretty sure this is not causing this problem.  

I have now totally disabled Windows Defender, which I don't need it anyway as have the excellent IMHO best PC AV system Kaspersky, but I have now also added I think all of the relevant AO folders to the Kaspersky exclusions list.  I have yet to try adding all teh XP12 folders to the exclusion list but will try this too.  

What I am trying to find is what is the actual common cause of this problem and that needs investigating to seek any commonality.  Maybe it is a combination of more than 1 issue but this sort of issue can be usually pinned down to one often simple cause.  

Can one of the AO developers here shed any light on this problem as they are surely the ones best equipped to fault trace this issue??  All these possible extraneous fixes are not getting to the root cause of the problem which must be some incompatibility that can be fixed in the AO coding without having to dumb down other XP12 features, and especially AV protection..

**Reply by malec** [reply]
I also configured sam to not add it's library automatically. I think that helped. Using xOrganizer (about $15) to manually add sam and make sure everything is in the right order. I will try to upgrade to 6 next.

**Reply by rayw45** [reply]
Well  I have completely disabled Windows  Defender in the WIndows registry and then ensured in Kaspersky that the complete dedicated XP12 SSD drive has been added to the exclusions, which would of course include everything in that drive including the massive AO cache and all the XP12 folders and files too, as well as I excluded the C: Drive Autoortho directory too.  Still the same problem so it is not an AV related problem it seems.  Don't know about SAM as I really do not want to lose my two airports and not sure how I can easily temporarily disable it. 

Now he has his AO working can malec try to re-enable say firstly SAM and check if AO still works okay, and then uninstall WInFSP and reinstall and enable Dokan to see if that changes anything with this debug issue??   We need to find exactly which measure malec took that actually did the trick as that is the usual logical fault finding process of elimination. Be good also if he can upgrade to ether the latest AO 6.1 too to see if that brings the problem back.

**Reply by malec** [reply]
I upgraded to 0.6.0-1 with the experimental installer and WinFSP and it works. Just ran the installer, renamed my old .autoortho files (to `v5.auto-ortho.bak`), restarted windows, ran & setup config, verified scenery order and started xplane.

**Reply by malec** [reply]
Also, SAM was always enabled, I just changed to this setting because the auto setting was changing my scenery order at xp runtime and breaking something:
![image](https://github.com/kubilus1/autoortho/assets/12704575/45c563e0-42b0-4118-a278-67015aa11325)

**Reply by rayw45** [reply]
Thanks for feedback Alec.  What XP12 plugins have you got running as now we
can eliminate them as the culprits.
What GFX card are you using too??

**Comment by rayw45** [comment]
Thanks for the feed back guys though have already tried many of these suggestions. Am always open to try other practical suggestions to locate the cause.

What I really would like to see if anyone needing this debug = True kludge fix has any items in their system spec that has commonality with my above listed spec which can be used as a baseline. I believe this may well help the developers here to locate and fix this annoying problem. For the record no I do not have RAID implemented on my system so maybe that can be eliminated as a possible cause and the problem persists whether of not I run an administrator.

I do not think using Python source code is a practical fix for 99% of users here and certainly not a path I want to go down.

**Reply by rayw45** [reply]
One other maybe important point I failed to mention is that I run X Plane from Steam.  So if anyone gets this debug needed problem not using Steram then we cam eliminate that factor.

**Reply by Jeff2122** [reply]
I have not noticed if the folks having this issue are all steam users. That would be something for sure.
I agree, using Python source is not for most folks, I am just wondering if it makes any difference in timing.

**Reply by rayw45** [reply]
So is there anyone here who does not run XP12 via Steam who has this debug = True problem too ?? As that should eliminate Steam as a culprit. 

Also wanted to add that I have a fast reliable1Gbps fibre optic internet connection which is as all such services should be and net neutral without any odious blocking or throttling other than some named porn sites :-0

**Reply by malec** [reply]
> So is there anyone here who does not run XP12 via Steam who has this debug = True problem too ?? As that should eliminate Steam as a culprit. 

I don't run via steam.

**Reply by rayw45** [reply]
But malec yours is working so it does not eliminate Steam of course.  I have just spent over 3 hours trying everything I can think of.  Disabling all XP12 plugins, completely temporarily switching off all AV, Disabling Global Forests 2, disabling all SImHeaven network overlays and enabling instead the AO overlay, all to try and find if anything will get this working without need for debug =True.  I now give in as nothing will work for me, so all plug ins I have installed on XP12 can be eliminated now, as can WinFSP or Dokan, as a cause it seems !!!!  

The problem must lie within AO 6.1 code itself as 4.4. works perfectly so what has changed that could cause the need for debug being enabled.  Why are some people having no problem in Windows 10??  Baffling and I hate it when I am unable to fix such an annoying problem with otherwise brilliant software.   Please can folk who have this problem confirm what GFX card they are using as maybe it is a NVidea and not Radeon problem??.  Same with CPU is it related to Intel and not AMD ??  

I was hoping for some input form the developers here on this problem that many are suffering from. as they surely have some sort of clue as to the cause.  So now I either give up on AO until this is fixed, put up with the big slow down with 6.1 and debug on, or go back to version 4.4 which works fine.
